src/checkers/inference/InferenceTreeAnnotator.java

```diff
@@ -160,6 +173,16 @@ public Void visitIdentifier(IdentifierTree node, AnnotatedTypeMirror identifierT
                     // another interface
                     variableAnnotator.visit(identifierType, node);
 
+                } else if (parentNode.getKind() == Kind.NEW_CLASS
+                        && ((NewClassTree) parentNode).getIdentifier() == node) {
+                    // This can happen at two locations in a NewClassTree:
+                    // (1) The type identifier of the NewClassTree, as `A` of `new @HERE A() {}`,
+                    // where `@HERE` is either an explicit annotation or a variable annotation.
```

Comment:
What does the `or a variable annotation` mean here? If there is already a `VarAnnot` for that location, why run the variableAnnotator again?